.github/pcb_pinmap.txt

```diff
@@ -0,0 +1,75 @@
+
+---- CS PCB v1.5 Pin Map:
+#define CONFIG_DCC_TRACK_SIGNAL_PIN 16
+#define CONFIG_OPS_HBRIDGE_DRV8873_5A 1
+#define CONFIG_OPS_TRACK_ENABLE_PIN 17
+#define CONFIG_PROG_HBRIDGE_DRV880X 1
+#define CONFIG_PROG_TRACK_ENABLE_PIN 18
+#define CONFIG_RAILCOM_TRIGGER_PIN 12
+#define CONFIG_RAILCOM_DATA_PIN 26
+#define CONFIG_RAILCOM_DIRECTION_PIN 25
+#define CONFIG_OLED_RESET_PIN 0
+#define CONFIG_FACTORY_RESET_PIN 32
+#define CONFIG_BOOTLOADER_PIN 33
+#define CONFIG_OLCB_TWAI_RX_PIN 4
+#define CONFIG_OLCB_TWAI_TX_PIN 5
+#define CONFIG_OPSTRACK_ADC_CHANNEL_0 1
+#define CONFIG_PROGTRACK_ADC_CHANNEL_3 1
+#define CONFIG_I2C_SCL_PIN 21
+#define CONFIG_I2C_SDA_PIN 23
+#define CONFIG_STATUS_LED_DATA_PIN 22
+#define CONFIG_TEMPSENSOR_ADC_CHANNEL_7 1
+#define CONFIG_RAILCOM_CUT_OUT_ENABLED 1
+#define CONFIG_RAILCOM_DATA_ENABLED 1
+
+---- CS PCB v1.0 Pin Map:
```

Comment:
@TrainzLuvr Can you review these pin mappings? I used the latest THT mappings as a starting point.

Reply:
IIRC I'm using PCB v1.3 pinouts because you've added some signals since and moved others. Plus the board is a bastardized version of v1.0 with backported features from v1.5, and daughter cards for errata. :)

[https://trainsluvr.com/for_forums/20201228-modded_esp32cs_v1.jpg](url)

Here's what I got, bolded items were not in your list, italic is what I removed from mine:

```
CONFIG_DCC_TRACK_SIGNAL_PIN 17
CONFIG_OPS_HBRIDGE_LMD18200 1
CONFIG_OPS_TRACK_ENABLE_PIN 4
**CONFIG_OPS_TRACK_ENABLED 1**
CONFIG_DCC_TRACK_BRAKE_PIN 26
_CONFIG_PROG_TRACK_ENABLE_PIN 18_
CONFIG_RAILCOM_TRIGGER_PIN 12
CONFIG_RAILCOM_DATA_PIN 26
CONFIG_RAILCOM_DIRECTION_PIN 25
CONFIG_OLED_RESET_PIN -1      
CONFIG_FACTORY_RESET_PIN -1
CONFIG_BOOTLOADER_PIN -1
CONFIG_OLCB_TWAI_RX_PIN 27
CONFIG_OLCB_TWAI_TX_PIN 32
CONFIG_OPSTRACK_ADC_CHANNEL_0 1
_CONFIG_PROGTRACK_ADC_CHANNEL_3 1_
CONFIG_I2C_SCL_PIN 21
CONFIG_I2C_SDA_PIN 23
CONFIG_STATUS_LED_DATA_PIN 22
CONFIG_TEMPSENSOR_ADC_CHANNEL_7 -1
CONFIG_RAILCOM_CUT_OUT_ENABLED 1 (don't have this)
CONFIG_RAILCOM_DATA_ENABLED 1 (don't have this)
```
I think I've also rerouted a pin for current sense (another bastardized circuit), but now I don't see it in my config so I might have turned it off.

Reply:
`CONFIG_OPS_TRACK_ENABLED 1` would come from sdkconfig.h and not this pinmap specific file.

`CONFIG_RAILCOM_CUT_OUT_ENABLED` and `CONFIG_RAILCOM_DATA_ENABLED` are new and replace `CONFIG_RAILCOM_FULL` and `CONFIG_RAILCOM_CUTOUT_ONLY` since menuconfig does not support using `select KEY` for choice entries.

I'll add a PCB 1.3.0 "OPS only" version based on your pin maps above. Can you preserve your sdkconfig file and do a test build with this update by replacing your sdkconfig with:
```
CONFIG_ESP32CS_PCB_V130_OPS_ONLY=y
```
If you set this *BEFORE* running menuconfig it should work correctly.

Reply:
The older versions will be removed in time. The v1.5.0 PCB version will be released in time. The v2.0 PCBs are more of a WIP using the esp32-s3 and is quite some time out still, I can't even find the esp32-s3 SoCs in any quantity yet.

Reply:
So what is available now to build, if someone was to do it (there's one fellow on OpenLCB interested).

Reply:
With parts shortages it is challenging to build any version of the PCBs. But the current v1.5 PCB is usable.

Reply:
@TrainzLuvr I've reached out to the user from the OpenLCB list and will work with him off-list.

I've also now dropped v1.0.0 from the config leaving only v1.3.0, v1.5.0 and v2.0.0. Over time I'd expect v1.3.0 to be dropped from public selection (but selectable via the above sdkconfig entry).

Reply:
Thanks for reaching out to him. Removing v1.0 leaves no THT PCBs and perhaps there should be a THT version available - not everyone is comfortable with SMT?

Reply:
SMT started with v1.5, I have an unpublished v1.4 PCB that is a mix of THT/SMT which can be machine assembled by JLCPCB (or similar) provided parts are available. The v1.5.2 PCB which I'm testing is primarily SMT and as long as it checks out I will publish it.

Reply:
I have a photo here of v1.3 you sent me which was also a THT/SMT mix, and since there was never a v1.2, it makes v1.0 the only THT edition though.
I'd rather see v1.0 fixed up and not retired so that there could be two options (THT and SMT). Perhaps THT for DIY-ers, and pre-populated SMT for everyone else.

Reply:
> I'd rather see v1.0 fixed up and not retired so that there could be two options (THT and SMT).

I don't have the KiCad files for the original v1.0 PCBs unfortunately. There are other shortcomings in the original v1.0 (through v1.3) designs as well.

Reply:
I only have the gerbers you sent me which I used to make the PCB. I know about v1.0 shortcomings, but it might be worth while fixing it all up for a new updated THT version. Many people are still afraid of soldering SMT, or don't want to bother.

Reply:
What's wrong with LMD18200, or do you mean for PRG/LCC out? I suppose we can look around for viable options...

Reply:
the LMD18200 was listed as EOL at some point, it may have just been one variant though, but the cost on it is pretty crazy. Mouser lists the LMD18200T/NOPB at $21.56 USD each! L298 is slightly better at $6.42 via Mouser.

I'd like to find something that has higher availability and better pricing if we can.